### PR TITLE
Closes #426 | Create dataset loader for CLIRMatrix

### LINKED ISSUE #426: Create dataset loader for CLIRMatrix
Dataloader name: `clir_matrix/clir_matrix.py`
DataCatalogue: http://seacrowd.github.io/seacrowd-catalogue/card.html?clir_matrix

|     Dataset| clir_matrix |
|-------------|---|
| Description | CLIRMatrix is a massively large collection of bilingual and multilingual datasets for Cross-Lingual Information Retrieval extracted automatically from Wikipedia. CLIRMatrix comprises (1) BI-139, a bilingual dataset of queries in one language matched with relevant documents in another language for 139x138=19,182 language pairs, and (2) MULTI-8, a multilingual dataset of queries and documents jointly aligned in 8 different languages. |
| Subsets | - |
| Languages | tgl, ilo, min, jav, sun, ceb, vie, tha |
| Tasks | Text Retrieval |
| License | Unknown (unknown) |
| Homepage | https://github.com/ssun32/CLIRMatrix |
| HF URL | - |
| Paper URL | https://aclanthology.org/2020.emnlp-main.340/ |

### PR DESCRIPTION
**Pull request by fhudi**
Closes #426

**_source-only_** dataloader.

### Checkbox
- [x] Confirm that this PR is linked to the dataset issue.
- [x] Create the dataloader script `seacrowd/sea_datasets/{my_dataset}/{my_dataset}.py` (please use only lowercase and underscore for dataset folder naming, as mentioned in dataset issue) and its `__init__.py` within `{my_dataset}` folder.
- [x] Provide values for the `_CITATION`, `_DATASETNAME`, `_DESCRIPTION`, `_HOMEPAGE`, `_LICENSE`, `_LOCAL`, `_URLs`, `_SUPPORTED_TASKS`, `_SOURCE_VERSION`, and `_SEACROWD_VERSION` variables.
- [x] Implement `_info()`, `_split_generators()` and `_generate_examples()` in dataloader script.
- [ ] Make sure that the `BUILDER_CONFIGS` class attribute is a list with at least one `SEACrowdConfig` for the source schema and one for a seacrowd schema.
- [x] Confirm dataloader script works with `datasets.load_dataset` function.
- [ ] Confirm that your dataloader script passes the test suite run with `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py` or `python -m tests.test_seacrowd seacrowd/sea_datasets/<my_dataset>/<my_dataset>.py --subset_id {subset_name_without_source_or_seacrowd_suffix}`.
- [x] If my dataset is local, I have provided an output of the unit-tests in the PR (please copy paste). This is OPTIONAL for public datasets, as we can test these without access to the data files.

---

Note:
- As this dataloader is source-only, seacrowd schema is not implemented.
- test was done using `tests.test_seacrowd_source_only.py`.

### CODE REVIEW

File: seacrowd/sea_datasets/clir_matrix/clir_matrix.py [hunk 1]
```diff
@@ -0,0 +1,234 @@
...
+}
+_URLS = {
+    ds: {
+        split: {(lque, ldoc): (f"https://www.cs.jhu.edu/~shuosun/clirmatrix/data/BI-139/{ds}/{_CLIR_LANG[lque]}/" f"{_CLIR_LANG[lque]}.{_CLIR_LANG[ldoc]}.{split}{'.base' if ds == 'base' else ''}.jl.gz") for lque, ldoc in permutations(_LANGUAGES, 2)}
+        for split in ["train", "dev", "test1", "test2"]
+    }
+    for ds in ["base", "full"]
+} | {"docs": {ldoc: f"https://www.cs.jhu.edu/~shuosun/clirmatrix/data/DOCS/{_CLIR_LANG[ldoc]}.tsv.gz" for ldoc in _LANGUAGES}}
+
+_SUPPORTED_TASKS = [Tasks.TEXT_RETRIEVAL]
```
[the remark this answers is not in the document]
Comment (fhudi) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py:83]: 🙇

File: seacrowd/sea_datasets/clir_matrix/clir_matrix.py [hunk 2]
```diff
@@ -0,0 +1,234 @@
...
+            datasets.SplitGenerator(
+                name=datasets.Split.TRAIN,
+                gen_kwargs={"filepath": data_paths["train"], "doc_paths": doc_paths},
+            ),
+            datasets.SplitGenerator(
+                name=datasets.Split.TEST,
+                gen_kwargs={"filepath": data_paths["test1"], "doc_paths": doc_paths},
+            ),
+            datasets.SplitGenerator(
+                name="test2",
```
[the remark this answers is not in the document]
Comment (fhudi) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py:195]: > ... any information regarding the `test2`'s difference ...

None that I can find from the publications, both the emnlp's paper or the homepage site.


> ... from the usual `test` split? ...

It looks the same to me.

---

@ssun32 
Can help to clarify?

File: seacrowd/sea_datasets/clir_matrix/clir_matrix.py [hunk 3]
```diff
@@ -0,0 +1,234 @@
...
+                version=datasets.Version(_SOURCE_VERSION),
+                description=f"{_DATASETNAME} source schema",
+                schema="source",
+                subset_id=f"{_DATASETNAME}{subset}",
+            )
+            for subset in [f"{'_' if lque else ''}{lque}{'_' if ldoc else ''}{ldoc}" for lque, ldoc in [("", ""), *permutations(_LANGUAGES, 2)]]
+        ],
+        *[
+            SEACrowdConfig(
+                name=f"{_DATASETNAME}{subset}_full_source",
```
[the remark this answers is not in the document]
Comment (fhudi) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py:109]: @holylovenia 
The former refers to the `base` split.

image

[reference page](https://www.cs.jhu.edu/~shuosun/clirmatrix/)

[the remark this answers is not in the document]
Comment (holylovenia) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py:109]: > @holylovenia The former refers to the `base` split.
> 
> image
> [reference page](https://www.cs.jhu.edu/~shuosun/clirmatrix/)

I see! It's very weird that the `full` < `base`. 😂 Could you please add this information to the description (appended) and to near the subset names (as a comment)?

[the remark this answers is not in the document]
Comment (fhudi) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py]: Done with the latest commit. Please have a look 🙏

File: seacrowd/sea_datasets/clir_matrix/clir_matrix.py (hunk 2 above)
[the remark this answers is not in the document]
Comment (holylovenia) [reply seacrowd/sea_datasets/clir_matrix/clir_matrix.py:195]: Thanks for the clarification, @ssun32! @fhudi, could you put this as a comment too?

**Review by holylovenia — APPROVED** [review]
Thanks for the addition, @fhudi! Aside from another comment request above, it works well. I'm approving this PR.

Let's wait for @luckysusanto's review.

**Comment by holylovenia** [comment]
Hi @muhammadravi251001, I assigned you in place of @luckysusanto due to the lack of response.

I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

**Comment by muhammadravi251001** [comment]
> Hi @muhammadravi251001, I assigned you in place of @luckysusanto due to the lack of response.
> 
> I would like to let you know that we plan to finalize the calculation of the open contributions (e.g., dataloader implementations) **in 31 hours**, so it'd be great if we could wrap up the reviewing and merge this PR before then.

Alright kak Holy, I will try my best. Maybe I will finish it at 2.00 PM on Jakarta time kak!

**Review by muhammadravi251001 — APPROVED** [review]
Approved. LGTM. Thanks for the contribution @fhudi!

**Pull request merged**